monocle/static/js/main.js

```diff
@@ -357,8 +357,23 @@ function getWorkers() {
     });
 }
 
-var map = L.map('main-map', {preferCanvas: true}).setView(_MapCoords, 13);
+var params = {};
+window.location.href.replace(/[?&]+([^=&]+)=([^&]*)/gi, function(m, key, value) {
+  params[key] = parseFloat(value);
+});
+
+var parsezoom = 13;
+if(parseFloat(params.zoom)) parsezoom = params.zoom;
 
+if(parseFloat(params.lat) && parseFloat(params.lon)){
+    var map = new L.Map('main-map', {
+        center: [params.lat, params.lon], 
```

Comment:
`parseFloat('12.12#')` will return `12.12` which is true, you should set the parsed values too, and not use it only to validate the imput

=> `Uncaught Error: Invalid LatLng object: (18.76470, 101.42516#)`